### PR TITLE
[catbuffer/parser] refactor some py generator code into the AST object model

### PR DESCRIPTION
**Pull request by Jaguar0625**
- eliminate most of type_objects and use ASTs directly

### CODE REVIEW

File: catbuffer/parser/catparser/ast.py [hunk 1]
```diff
@@ -339,6 +388,36 @@ def __init__(self, tokens, disposition=None):
...
+	@property
+	def display_type(self):
+		"""Gets the display type."""
+		return DisplayType.UNSET if isinstance(self.field_type, str) else self.field_type.display_type
+
+	@property
+	def size(self):
+		"""Gets the backing size."""
+		return self.field_type.size if hasattr(self.field_type, 'size') else None
+
```
Comment (gimre-xymcity) [inline catbuffer/parser/catparser/ast.py:420]: this whole piece with properties looks just great (I mean the fact that it looks like cosmetic compared to old)

File: sdk/python/generator/StructTypeFormatter.py [hunk 2]
```diff
@@ -136,12 +136,13 @@ def generate_condition(self, field, prefix_field=False):
...
 		value = f'{conditional.value}'
-		yoda_value = value if condition_field.get_type().is_int else f'{condition_field.get_type().typename}.{value}'
+		condition_ast_model = condition_field.type_ast_model
+		yoda_value = value if DisplayType.INTEGER == condition_ast_model.display_type else f'{condition_ast_model.name}.{value}'
 		field_prefix = 'self.' if prefix_field else ''
 
 		# HACK: instead of handling dumb magic value in namespace parent_name, generate slightly simpler condition
-		if prefix_field and isinstance(field.ast_model.field_type, Array):
-			return f'if {field_prefix}{field.original_field_name}:\n'
+		if prefix_field and DisplayType.UNSET != field.ast_model.display_type:
```
Comment (gimre-xymcity) [inline sdk/python/generator/StructTypeFormatter.py:144]: this looks bit too wide? (I assume it works correctly though)

**Review by gimre-xymcity — APPROVED** [review]
awesome

File: sdk/python/generator/StructTypeFormatter.py (hunk 2 above)
Comment (Jaguar0625) [reply sdk/python/generator/StructTypeFormatter.py:144]: possibly, but it works so would leave it for now

**Pull request merged**